consensus/src/main/java/org/ethereum/beacon/consensus/SpecHelpers.java

```diff
@@ -1259,22 +1434,36 @@ public void checkShardRange(ShardNumber shard) {
     return publicKeys;
   }
 
-  // def lmd_ghost(store, start):
-  //    validators = start.state.validator_registry
-  //    active_validators = [validators[i] for i in
-  //                         get_active_validator_indices(validators, start.state.slot)]
-  //    attestation_targets = [get_latest_attestation_target(store, validator)
-  //                           for validator in active_validators]
-  //    def get_vote_count(block):
-  //        return len([target for target in attestation_targets if
-  //                    get_ancestor(store, target, block.slot) == block])
-  //
-  //    head = start
-  //    while 1:
-  //        children = get_children(head)
-  //        if len(children) == 0:
-  //            return head
-  //        head = max(children, key=get_vote_count)
+  /*
+    def lmd_ghost(store: Store, start_state: BeaconState, start_block: BeaconBlock) -> BeaconBlock:
+      """
+      Execute the LMD-GHOST algorithm to find the head ``BeaconBlock``.
+      """
+      validators = start_state.validator_registry
+      active_validators = [
+          validators[i]
+          for i in get_active_validator_indices(validators, start_state.slot)
+      ]
+      attestation_targets = [
+          get_latest_attestation_target(store, validator)
+          for validator in active_validators
+      ]
+
+      def get_vote_count(block: BeaconBlock) -> int:
+          return len([
+              target
+              for target in attestation_targets
+              if get_ancestor(store, target, block.slot) == block
```

Comment:
`get_ancestor` was updated too from if-else to if-elif-else:
```
Let get_ancestor(store: Store, block: BeaconBlock, slot: SlotNumber) -> BeaconBlock be the ancestor of block with slot number slot. The get_ancestor function can be defined recursively as:
def get_ancestor(store: Store, block: BeaconBlock, slot: SlotNumber) -> BeaconBlock:
    """
    Get the ancestor of ``block`` with slot number ``slot``; return ``None`` if not found.
    """
    if block.slot == slot:
        return block
    elif block.slot < slot:
        return None
    else:
        return get_ancestor(store, store.get_parent(block), slot)
```
proof https://github.com/ethereum/eth2.0-specs/blob/4cdb66728a27739fcb35e545386770f3a2972bf8/specs/core/0_beacon-chain.md

Reply:
Good catch!